src/core/crypto/mbedtls.cpp

```diff
@@ -42,6 +42,7 @@
 namespace ot {
 namespace Crypto {
 
+#ifdef MBEDTLS_MEMORY_BUFFER_ALLOC_C
```

Comment:
Isn't the logic inverted in here, shouldn't it be `#ifndef MBEDTLS_MEMORY_BUFFER_ALLOC_C`?

`MBEDTLS_MEMORY_BUFFER_ALLOC_C` allows to configure mbedTLS to use it's own internal memory allocator, once we provide it with some memory to manage, see https://tls.mbed.org/kb/how-to/using-static-memory-instead-of-the-heap. Here, it was used to enable OpenThread's memory allocator.

I'm asking this in reference to #3578. We use `MBEDTLS_MEMORY_BUFFER_ALLOC_C` for memory management in Zephyr, so I wanted to test this change,  yet currently it still uses OpenThread heap instead.

Reply:
@rlubos I added a specific macro for this purpose.

Reply:
Thanks, appreciate that. I'm wondering, would it make sense to make this config a little more generic, let's say `OPENTHREAD_CONFIG_DISABLE_MBEDTLS_MANAGEMENT`? This could cover mbedTLS log level configuration as well:
https://github.com/openthread/openthread/blob/749421641513cc71bfc9bb899c5b683d6ce060c9/src/core/meshcop/dtls.cpp#L179
It should cover then all global mbedTLS configurations, others are context-specific. Not sure if it's not in conflict with your usecase though.

Reply:
Shall we just prevent OpenThread managing mbedTLS when builtin-mbedtls is disabled? @jwhui

Reply:
I was thinking about this originally, yet I came to conclusion that these two use-cases are not necesirally tightly coupled. This scenario should work for me in Zephyr, though I'm not sure if it won't broke something for other users of this option.